src/test/java/edu/ucsb/cs156/example/controllers/UCSBDatesControllerTests.java

```diff
@@ -0,0 +1,313 @@
+
+                // act
+                MvcResult response = mockMvc.perform(get("/api/ucsbdates?id=7"))
+                                .andExpect(status().isOk()).andReturn();
+
+                // assert
+
+                verify(ucsbDateRepository, times(1)).findById(eq(7L));
+                String expectedJson = mapper.writeValueAsString(ucsbDate);
+                String responseString = response.getResponse().getContentAsString();
+                assertEquals(expectedJson, responseString);
+        }
+
+        @WithMockUser(roles = { "USER" })
+        @Test
+        public void test_that_logged_in_user_can_get_by_id_when_the_id_does_not_exist() throws Exception {
+
+                // arrange
+
+                when(ucsbDateRepository.findById(eq(7L))).thenReturn(Optional.empty());
+
+                // act
+                MvcResult response = mockMvc.perform(get("/api/ucsbdates?id=7"))
+                                .andExpect(status().isBadRequest()).andReturn();
+
+                // assert
+
+                verify(ucsbDateRepository, times(1)).findById(eq(7L));
+                String responseString = response.getResponse().getContentAsString();
+                assertEquals("UCSBDate with id 7 not found", responseString);
+        }
+
+        @WithMockUser(roles = { "USER" })
+        @Test
+        public void logged_in_user_can_get_all_ucsbdates() throws Exception {
+
+                // arrange
+                LocalDateTime ldt1 = LocalDateTime.parse("2022-01-03T00:00:00");
+
+                UCSBDate ucsbDate1 = UCSBDate.builder()
+                                .name("firstDayOfClasses")
+                                .quarterYYYYQ("20222")
+                                .localDateTime(ldt1)
+                                .build();
+
+                LocalDateTime ldt2 = LocalDateTime.parse("2022-03-11T00:00:00");
+
+                UCSBDate ucsbDate2 = UCSBDate.builder()
+                                .name("lastDayOfClasses")
+                                .quarterYYYYQ("20222")
+                                .localDateTime(ldt2)
+                                .build();
+
+                ArrayList<UCSBDate> expectedDates = new ArrayList<>();
+                expectedDates.addAll(Arrays.asList(ucsbDate1, ucsbDate2));
+
+                when(ucsbDateRepository.findAll()).thenReturn(expectedDates);
+
+                // act
+                MvcResult response = mockMvc.perform(get("/api/ucsbdates/all"))
+                                .andExpect(status().isOk()).andReturn();
+
+                // assert
+
+                verify(ucsbDateRepository, times(1)).findAll();
+                String expectedJson = mapper.writeValueAsString(expectedDates);
+                String responseString = response.getResponse().getContentAsString();
+                assertEquals(expectedJson, responseString);
+        }
+
+        @WithMockUser(roles = { "ADMIN" })
```

Comment:
Can we start using `@WithMockUser(roles = { "ADMIN", "USER" })` for these tests? For these particular tests, it doesn't matter, but a student encountered an issue where USER authorized routes don't work in tests if the user only has the role ADMIN and not USER as well. It might help to mitigate that issue if we set a precedent of giving mock admins both roles in tests where we're mocking an admin.